artemis-commons/src/main/java/org/apache/activemq/artemis/utils/collections/TypedProperties.java

```diff
@@ -931,30 +954,75 @@ protected StringValue create(final ByteBuf byteBuf, final int length) {
          }
       }
 
-      final SimpleString val;
+      public static final class StringStringValuePool extends AbstractPool<String, StringValue> {
+
+         public StringStringValuePool() {
+            super();
+         }
+
+         public StringStringValuePool(final int capacity) {
+            super(capacity);
+         }
+
+         @Override
+         protected StringValue create(String value) {
+            return new StringValue(SimpleString.toSimpleString(value));
+         }
+
+         @Override
+         protected boolean isEqual(StringValue entry, String value) {
+            return false;
```

Comment:
I'm not getting this `false` here!! That means that the interner will always replace the interned entry?

Reply:
it was a miss, good catch, fixed.